src/Windows/Host/Client/Impl/BrokerServices/LocalStaticFileServer.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.IO;
+using Microsoft.Common.Core.Logging;
+
+namespace Microsoft.R.Host.Client.BrokerServices {
+    public class LocalStaticFileServer : StaticFileServerBase {
+        private static LocalStaticFileServer _server;
+        private static object _serverLock = new object();
+
+        public LocalStaticFileServer(IFileSystem fs, IActionLog log, IConsole console) : base(fs, log, console) {
+        }
+
+        public async Task<string> HandleUrlAsync(string urlStr, CancellationToken ct = default(CancellationToken)) {
+            await InitializeAsync(ct);
+
+            Log.WriteLine(LogVerbosity.Minimal, MessageCategory.General, Resources.Info_StaticFileServerStarted.FormatInvariant(Listener.Prefixes.FirstOrDefault()));
+            Console.WriteLine(Resources.Info_StaticFileServerStarted.FormatInvariant(Listener.Prefixes.FirstOrDefault()));
```

Comment:
Console here is a property that provides the saved IConsole reference. In this case it goes to REPL.